### PR TITLE
[Compile Time Constant Extraction] Add `@extractConstantsFromMembers`

### PR DESCRIPTION
**Pull request by ApolloZhu**
Add `@extractConstantsFromMembers` attribute under `ExtractConstantsFromMembers` experimental feature flag

### CODE REVIEW

**Comment by ApolloZhu** [comment]
@swift-ci please smoke test

**Comment by ApolloZhu** [comment]
@swift-ci please smoke test

**Review by artemcm — APPROVED** [review]

File: test/Parse/extractConstantsFromMembers_attr.swift [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// RUN: %target-typecheck-verify-swift -parse -enable-experimental-feature ExtractConstantsFromMembers
```
Comment (artemcm) [inline test/Parse/extractConstantsFromMembers_attr.swift:1]: Could you please add a small lit test to ensure we emit `attr_extractConstantsFromMembers_experimental` when not enabled with the opt-in flag?

**Comment by ApolloZhu** [comment]
@swift-ci please test

File: test/Parse/extractConstantsFromMembers_attr.swift (hunk 1 above)
Comment (ApolloZhu) [reply test/Parse/extractConstantsFromMembers_attr.swift:1]: Good catch, I've added a test case for this

**Review by artemcm — APPROVED** [review]

**Comment by compnerd** [comment]
Is there an evolution proposal that went along with this?  If this is not yet approved, should this not be spelt with a leading `_`?

**Comment by hborla** [comment]
@compnerd SE-0362 standardized the in-development language feature model to use experimental feature flags instead of (or in addition to, depending on the circumstance) the `_` attribute spelling: https://github.com/apple/swift-evolution/blob/main/proposals/0362-piecemeal-future-features.md#embracing-experimental-features

**Comment by compnerd** [comment]
@hborla might be good to be more explicit about that in the prose.  I never realized that was part of the proposal.

**Comment by hborla** [comment]
@compnerd That's fair, and we should also include that in the Swift Evolution process document. I'm not sure it's been updated to include mention of experimental features since SE-0362 has been accepted. I'll make that change 👍🏻

**Pull request merged**